pkg/controllers/user/globaldns/user_cluster_ingress.go

```diff
@@ -116,59 +117,74 @@ func (ic *UserIngressController) noGlobalDNS() bool {
 	return len(gd) == 0
 }
 
-func gatherIngressEndpoints(ingressEps []v1.LoadBalancerIngress) []string {
-	endpoints := []string{}
-	for _, ep := range ingressEps {
-		if ep.IP != "" {
-			endpoints = append(endpoints, ep.IP)
-		} else if ep.Hostname != "" {
-			endpoints = append(endpoints, ep.Hostname)
+func (ic *UserIngressController) reconcileAllGlobalDNSs() (runtime.Object, error) {
+	globalDNSList, err := ic.globalDNSLister.List("", labels.NewSelector())
```

Comment:
Global dns object belongs to `cattle-global-data` ns, so it we would make things faster if we pass the ns in

Reply:
Can we revise other places in the code to check where else the ns needs to be passed

Reply:
revised with the ns